Main.py

```diff
@@ -1,26 +1,71 @@
-from Register import Register
+from QuantumRegister import *
 from Qubit import Qubit
 import Operators as op
-import Tests
 
-# this main runs through a few of the materials covered in the notes
-# the three seven tests is covered in the slides
-print("REGISTER STATE: VECTOR AS VERTICAL MATRIX")
-r1 = Tests.threeSevenTest()
-Tests.normalizedTest(r1)
-r1.updateS()
+import deutsch
+import deutschJozsa
+import grovers
 
-# these tests are to ensure the hadamard is taking form that is appropriate 
-# it seems to be working as intended
 
-# the register must always be normalised, 
-# a register involving multiple states of fractions is in a superposition
-r2 = Tests.hadamardTest()
-Tests.normalizedTest(r2)
-Tests.stateTest()
-Tests.hadAdvanced()
 
+def deutschMain():
+	# This function is used to run the deutsch algorithm, it uses examples for both balanced and constant functions
+	# The deutsch algorithm is used to evaluate a black box algorithm using a basic quantum circuit, for all constant 
+	# functions our implementation will produce a 0 and for all balanced, a 1
 
-r2.updateE()
+	def balanced(x):
+		return x % 2
+	def constant(x):
+		return 0

```

Comment:
Checkout new oracle implementation, oracle functions should take in a register and return it. Allows us to build oracles out of gates as well